core/variant/variant_setget.cpp

```diff
@@ -1911,6 +1911,145 @@ Variant Variant::recursive_duplicate(bool p_deep, int recursion_count) const {
 	}
 }
 
+void Variant::add(const Variant &a, const Variant &b, Variant &r_dst) {
+	if (a.type != b.type) {
+		return;
+	}
+
+	switch (a.type) {
+		case NIL: {
+			r_dst = Variant();
+		}
+			return;
+		case INT: {
+			int64_t va = a._data._int;
+			int64_t vb = b._data._int;
+			r_dst = int(va + vb);
+		}
+			return;
+		case FLOAT: {
+			double ra = a._data._float;
+			double rb = b._data._float;
+			r_dst = ra + rb;
+		}
+			return;
+		case VECTOR2: {
+			r_dst = *reinterpret_cast<const Vector2 *>(a._data._mem) + *reinterpret_cast<const Vector2 *>(b._data._mem);
+		}
+			return;
+		case VECTOR2I: {
+			int32_t vax = reinterpret_cast<const Vector2i *>(a._data._mem)->x;
+			int32_t vbx = reinterpret_cast<const Vector2i *>(b._data._mem)->x;
```

Comment:
Instead of doing multiple reinterpret casts to access all members, you could just do
```C++
Vector2i va = a.operator Vector2i();
```
and then use `va` directly. There is just lots of useless casting in your code.